lib/moon/components/input_group/input_group.ex

```diff
@@ -0,0 +1,109 @@
+defmodule Moon.Components.InputGroup do
+  @moduledoc false
+
+  use Moon.StatelessComponent
+  alias Moon.Components.InputGroup.Container
+
+  prop orientation, :string,
+    default: "horizontal",
+    values: ["horizontal", "vertical"]
+
+  prop dir, :string, default: "ltr", values: ["ltr", "rtl"]
+  prop background_color, :string, default: "gohan-100", values: Moon.colors()
+  prop has_fields, :boolean, default: false
+
+  slot default, args: [:input_group_class], required: true
+
+  def render(assigns) do
+    ~F"""
+    <Context
+      put={__MODULE__, orientation: @orientation}
+      put={__MODULE__, dir: @dir}
+      put={__MODULE__, background_color: @background_color}
+      put={__MODULE__, is_in_input_group: true}
+    >
+      <Container>
+        <#slot :args={input_group_class: get_class_input_group(@has_fields, @orientation, @dir)} />
```

Comment:
Please do not use slot args. 

You are using Context incorrectly. When you put here value into context:

`put={__MODULE__, orientation: @orientation}`

Then you can read it

`get={Moon.Components.InputGroup, orientation: orientation}`

but you try to read it

`get={Moon.Components.TextInput, orientation: orientation}`

and that is why you claim that contexts do not work. :)

Reply:
I use the correct module, `get={Moon.Components.InputGroup, orientation: orientation}` but it still does not work because the nested slots cannot access the context values. I filed this issue: [https://github.com/surface-ui/surface/issues/639](here) Have you tried it?